core/cppgc.rs

```diff
@@ -18,6 +18,80 @@ struct CppGcObject<T: GarbageCollected> {
   member: T,
 }
 
+/// PrototypeChain defines the position of the object in the prototype chain.
+///
+///   A    (index 0)
+///  / \
+/// B   C  (index 1)
+/// |
+/// D      (index 2)
+///
+/// where each B, C and D hold their own prototype chain.
+pub trait PrototypeChain {
+  fn prototype_index() -> Option<usize> {
+    None
+  }
+}
+
+/// A good enough number for the maximum prototype chain depth based on the
+/// usage.
+const MAX_PROTO_CHAIN: usize = 3;
```

Comment:
Are you sure we're not gonna hit this limit immediately in Deno?

Reply:
Nope, this is only for native object proto chain. The actual JS proto chain isn't restricted.